### PR TITLE
Rebalance Fee Generalization

### PR DESCRIPTION
**Pull request by felix2feng**

### CODE REVIEW

File: contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol [hunk 1]
```diff
@@ -240,15 +252,4 @@ contract RebalancingSetState {
     {
         return _tokenAddress == address(currentSet);
     }
-
-    /*
-     * Get array version of failedAuctionWithdrawComponents
-     */
-    function getFailedRebalanceComponents()
```
Comment (bweick) [inline contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol]: Why remove this?

File: contracts/core/fee-calculators/FixedFeeCalculator.sol [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+     *
+     * @param _feeCalculatorData       Bytes encoded data about the fee for the sender
+     */
+    function initialize(
+        bytes calldata _feeCalculatorData
+    )
+        external
+    {
+        // Sender fee must not already be initialized
+        require(
```
Comment (bweick) [inline contracts/core/fee-calculators/FixedFeeCalculator.sol]: Just put this all on a line

File: contracts/core/tokens/RebalancingSetTokenV2Factory.sol [hunk 3]
```diff
@@ -153,84 +161,71 @@ contract RebalancingSetTokenV2Factory {
...
         require(
-            address(parameters.liquidator) != address(0),
-            "Create: Invalid liquidator address"
+            address(parameters.liquidator) != address(0) && 
+            liquidatorWhitelist.whiteList(address(parameters.liquidator)),
+            "Invalid liquidator"
         );
 
-        // Require that liquidator is whitelisted by the liquidatorWhitelist
+        // Require liquidator address is non-zero and is whitelisted by the liquidatorWhitelist
```
Comment (bweick) [inline contracts/core/tokens/RebalancingSetTokenV2Factory.sol]: Duped comment this should be feeCalculator

File: contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol [hunk 4]
```diff
@@ -201,6 +201,18 @@ contract RebalancingSetState {
 
     /* ============ Getter Functions ============ */
 
+    /*
+     * Retrieves the current expected fee from the fee calculator
+     * Value is returned as a scale decimal figure.
+     */
+    function rebalanceFee()
```
Comment (bweick) [inline contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol]: Hmmm won't there be times where this doesn't return a number? For a constant fee it works but say the fee calculation requires knowing what the nextSet will be?

Comment (felix2feng) [reply contracts/core/tokens/rebalancing-v2/RebalancingSetState.sol]: In other implementations, this function could revert or return an anticipated value

**Pull request merged**